src/UrlHelper.php

```diff
@@ -28,21 +30,17 @@ public function __construct(BaseHelper $helper)
     /**
      * Proxies to `Zend\Expressive\Helper\UrlHelper::generate()`
      *
-     * @param null|string $routeName
-     * @param array $routeParams
-     * @param array $queryParams
-     * @param null|string $fragmentIdentifier
      * @param array $options Can have the following keys:
      *     - router (array): contains options to be passed to the router
      *     - reuse_result_params (bool): indicates if the current RouteResult
      *       parameters will be used, defaults to true
      * @return string
      */
     public function __invoke(
-        $routeName = null,
+        ?string $routeName = null,
         array $routeParams = [],
         array $queryParams = [],
-        $fragmentIdentifier = null,
+        ?string $fragmentIdentifier = null,
```

Comment:
as above

Reply:
Same rationale as previous; in this case, you could pass `null` if you want to provide router options, but not change the fragment.